fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/TemplateDruidQuery.java

```diff
@@ -424,9 +449,181 @@ private int calculateDepth(TemplateDruidQuery candidate) {
+                .filter(agg -> agg.getName().equals(currentName))
                 .findFirst()
+                .orElseThrow(IllegalArgumentException::new)
+                .withName(newName);
+
+        Set<Aggregation> newAggs = getAggregations().stream()
+                .filter(agg -> !agg.getName().equals(currentName))
+                .collect(Collectors.toSet());
+        newAggs.add(renamedAgg);
+
+        Set<PostAggregation> newPostAggs = new HashSet<>();
+        for (PostAggregation pa : getPostAggregations()) {
+            if (pa instanceof WithFields) {
+                newPostAggs.add(renamePostAggregationTree(currentName, renamedAgg, (WithFields<?>) pa));
+            } else if (pa instanceof AggregationReference) {
+                newPostAggs.add(renameAggregationReference(currentName, renamedAgg, (AggregationReference<?>) pa));
+            } else {
+                newPostAggs.add(pa);
+            }
+        }
+
+        return this
+                .withAggregations(newAggs)
+                .withPostAggregations(newPostAggs);
+    }
+
+    /**
+     * Updates all {@link PostAggregation}s that reference the {@link Aggregation} with output name that matches
+     * {@code oldFieldName} to point to the {@code renamedAgg}.
+     * <p>
+     * The behavior of this method when any parameter value is null is undefined. As such this method should never be
+     * called with null parameter values.
+     *
+     * @param oldFieldName  The name of the Aggregation that has been renamed. References to this Aggregation must be
+     *                      updated.
+     * @param renamedAgg  The new Aggregation that has been renamed from {@code oldFieldName}. All references to the
+     *                    previous Aggregations are repointed to this Aggregation.
+     * @param rootPostAgg  A PostAggregation that depends on other PostAggregations. If any of those dependent
+     *                     PostAggregations reference the original Aggregation, they are repointed at the new
+     *                     Aggregation.
+     * @return a PostAggregation that has had all of its dependent PostAggregations updated.
+     */
+    protected PostAggregation renamePostAggregationTree(
+            String oldFieldName,
+            Aggregation renamedAgg,
+            WithFields<? extends PostAggregation> rootPostAgg
+    ) {
+        List<PostAggregation> newFields = new ArrayList<>();
+        for (PostAggregation field : rootPostAgg.getFields()) {
+            PostAggregation newField;
+            if (field instanceof WithFields) {
+                newField = renamePostAggregationTree(oldFieldName, renamedAgg, (WithFields<?>) field);
+            } else if (field instanceof AggregationReference) {
+                newField = renameAggregationReference(oldFieldName, renamedAgg, (AggregationReference<?>) field);
+            } else {
+                newField = field;
+            }
+            newFields.add(newField);
```

Comment:
please add inline comments here to explain the syntactic meaning.  Something like 
above 568:  // This postagg has more than one dependency, recurse to visit each one.
above 570:  //This postagg is a leaf that references an aggregation
above else // This post agg does not reference dependencies and is unchanged.